setup.py

```diff
@@ -42,6 +42,9 @@
     "fuzz" : [
         "unicornafl>=2.0.0;platform_system=='Windows'",
         "fuzzercorn>=0.0.1;platform_system=='Linux'"
+    ],
+    "SCA" : [
+        "r2libr"
```

Comment:
@elicn maybe we could add `r2libr` as an essential dep? My thought roughly is:

- First of all, unlike the previously discussed `udbserver`, `r2libr` is maintained by me, in other words, by Qiling team directly.
- `r2libr` could be tightly integrated into our loader.
- The `r2libr` could help us polish some old API like `set_api` with the additional information.
- There seems no use case that users really would like to use QIling without `r2libr`?

Reply:
Also note, `radare2` sometimes introduces break changes. To specify `r2libr` please use something like `r2libr==5.7.0`.

Reply:
Breaking changes should only happen when the major and minor numbers change. only patch releases are abi stable. so if you have r2libr-5.7.0 you can use r2-5.7.0, but also 5.7.4 or 5.7.8 without runtime problems. That relaxes a little the stress to maintain updates and also makes it easy to transition between abi-breaking versions

Reply:
IMHO, I would refrain from defining something as a prerequisit unless it is absolutley required; that is, the program cannot operate without it. Plugging r2 into Qiling sounds like a great idea, where Qiling can benefit a lot from r2 static analysis.

Integrating r2 static analysis into Qiling requires some thinking of how to do it right and consistent across all modules. I really don't want it to be integrated into some modules and not in others. Until r2 is integrated across all Qiling modules, I believe we can wait with this prerequisit.

Reply:
Yes, I understand your concern so let me illustrate my motivation a bit more:

1. Radare2 helps reduce redundant code and improves robustness, e.g. ELF loader, PE loader, even disassembler etc.
2. Indeed, I strongly agree with you that r2 should be, some sort of `everywhere` in Qiling. I hope to achieve this by using r2 to build a few basic utils or modules. An example I showed before is to identify function boundaries. Imagine you can `set_api` on a binary without any symbol tables, which is really useful and I don't think users will reject it.

Based on these two points, I think r2 will become a prerequisite someday but also I agree we should wait until we really have to do so.